packages/core/protocols/src/errors.ts

```diff
@@ -78,3 +78,49 @@ export class RpcNotOpenError extends Error {
     super('RPC has not been opened.');
   }
 }
+
+export class WrappableError extends Error {
```

Comment:
FYI that I moved all the `@dxos/errors` here https://github.com/dxos/dxos/pull/4138/files#diff-1f72a0497a21cbb885fdee03300127134c6b2002ea75430e83a15d79805e0ee4 as well in order to be able to have them bridge the RPC boundary. We'll likely want to reconcile the error classes when merging.

Reply:
yeah, was uncertain how many of the protocol errors would be external-facing, but it's probably simpler to just have them all together.

btw i saw the `this.name` pattern suggested but it didn't seem to work with `ErrorStream` either, so back to string matching in https://github.com/dxos/dxos/pull/4139/commits/4edac01a97201893aa86bddbf97c3f7ac425a03e#diff-8fd5aaf7614ff1002b55a9d55308ab71bf42144d7246e3c8d68852c93cfc8745R185

Reply:
@wittjosiah thanks for that heads up, moved my errors to `packages/sdk/errors/src/errors.ts` and use `BaseError`

Reply:
after using this for a bit it feels a bit unnatural to have to import a separate package just for errors, i think it'd feel more natural as a consumer to import the errors from the same package of the code that produces them.

Reply:
@nferch on the PR above I moved all the errors to be defined in `@dxos/protocols` and they're all re-exported by `@dxos/client` (which is where I imagine most people would encounter them)

Reply:
@wittjosiah makes sense, so just more of a personal/internal concern then.